Lib/fontbakery/codetesting.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+# Copyright 2020 The Fontbakery Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+def get_check(profile, checkid):
+    def _checker(value):
+        from fontTools.ttLib import TTFont
+        from fontbakery.fonts_profile import execute_check_once
+        if isinstance(value, str):
+            return execute_check_once(profile, checkid, value)
+
+        elif isinstance(value, TTFont):
+            return execute_check_once(profile, checkid, value.reader.file.name,
+                                      {'ttFont': value})
+#
+# TODO: I am not sure how to properly handle these cases:
```

Comment:
I agree, I'll look into this. Especially because `ttFonts` is a "derived_iterable" and that's not done with a cache override of `conditions`.